services/DatastoreService.go

```diff
@@ -0,0 +1,165 @@
+package services
+
+import (
+	"context"
+	"github.com/onego-project/onego/blueprint"
+	"github.com/onego-project/onego/requests"
+	"github.com/onego-project/onego/resources"
+)
+
+// DatastoreService structure to manage OpenNebula datastore
+type DatastoreService struct {
+	Service
+}
+
+const enable = true
+
+// Allocate to create a new datastore in OpenNebula
+func (ds *DatastoreService) Allocate(ctx context.Context, blueprint blueprint.Interface, cluster resources.Cluster) (*resources.Datastore, error) {
+	clusterID, err := cluster.ID()
+	if err != nil {
+		return nil, err
+	}
+
+	blueprintText, err := blueprint.Render()
+	if err != nil {
+		return nil, err
+	}
+
+	resArr, err := ds.call(ctx, "one.datastore.allocate", blueprintText, clusterID)
+	if err != nil {
+		return nil, err
+	}
+
+	return ds.RetrieveInfo(ctx, int(resArr[resultIndex].ResultInt()))
+}
+
+// Delete to delete the given datastore from the pool
+func (ds *DatastoreService) Delete(ctx context.Context, dataStore resources.Datastore) error {
```

Comment:
The main structure is called `Datastore`. There is no need to call parameter `dataStore` (camel case).